tests/test_CharArrayNullTerm.cpp

```diff
@@ -0,0 +1,99 @@
+#define BOOST_TEST_MODULE Aggregate_Well_Data
+
+#include <boost/test/unit_test.hpp>
+
+#include <opm/output/eclipse/CharArrayNullTerm.hpp>
+
+// =====================================================================
+
+BOOST_AUTO_TEST_SUITE(AChar8)
+
+BOOST_AUTO_TEST_CASE (Basic_Operations)
+{
+    // Default Constructor
+    {
+        const auto s = Opm::RestartIO::Helpers::CharArrayNullTerm<8>{};
+
+        BOOST_CHECK_EQUAL(s.c_str(), std::string(8, ' '));
+    }
+
+    // Construct from Constant String
+    {
+        const auto s = Opm::RestartIO::Helpers::CharArrayNullTerm<8>{"Inj-1"};
```

Comment:
You are allowed to define a convenience alias for this type, and it would make it more readable I think.